docs/specs/sequences/endpoint_reconfiguration.md

```diff
@@ -0,0 +1,142 @@
+[**UDMI**](../../../) / [**Docs**](../../) / [**Specs**](../) / [**Sequences**](./) / [Endpoint Reconfiguration](#)
+
+# Endpoint Reconfiguration
+
+Endpoint reconfiguration is the reconfiguration of the UDMI endpoint through the
+UDMI blob delivery mechanisms via UDMI config messages.
+
+The [endpoint configuration blob](https://github.com/faucetsdn/udmi/blob/master/tests/configuration_endpoint.tests/simple.json) is a JSON object defined by
+[configuration_endpoint.json](https://faucetsdn.github.io/udmi/gencode/docs/configuration_endpoint.html), which is base64 encoded in the config message.
+
+
+## Tests
+
+### Valid Endpoint (Successful) Reconfiguration
+
+Note - `<ENDPOINT>` is a **base64** encoded endpoint object
+
+```mermaid
+%%{wrap}%%
+sequenceDiagram
+    autonumber
+    participant D as Device
+    participant E as Original Endpoint
+    participant E' as New Endpoint
+    E->>D:CONFIG MESSAGE<br>blobset.blobs._iot_endpoint_config.base64 = <ENDPOINT><br>blobset.blobs._iot_endpoint.phase = "final"
+    D->>E:STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.phase = "apply"
+    D-->>E':CONNECTION ATTEMPT
+    E'->>D:CONFIG MESSAGE
+    D->>E':STATE MESSAGE<BR>blobset.blobs._iot_endpoint_config.phase = "final"
+```
+
+### Invalid Endpoint (Unsuccessful Reconfiguration)
```

Comment:
probably not something to do now, but something to think about, is that there's another category of "invalid" here which is an invalid request -- i.e. the base64 is borked or the resulting JSON is not parsable.  In a perfect world that would be just another test and then another error category (which is semantically different than "we understood the request but that failed")...  Similar to writeback. There are clear stages where this is different so I think we can clearly separate out the tests.  (E.g. to make sure the JSON parser doesn't barf and bring the system down!)